components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/dao/test/GatewayArtifactsMgtDAOTest.java

```diff
@@ -0,0 +1,149 @@
+    ByteArrayInputStream anyInputStream = new ByteArrayInputStream("test data".getBytes());
+
+    @Before
+    public void setUp() throws Exception {
+        String dbConfigPath = System.getProperty("APIManagerDBConfigurationPath");
+        APIManagerConfiguration config = new APIManagerConfiguration();
+        initializeDatabase(dbConfigPath);
+        config.load(dbConfigPath);
+        ServiceReferenceHolder.getInstance().setAPIManagerConfigurationService(new APIManagerConfigurationServiceImpl
+                (config));
+        ServiceReferenceHolder.getInstance().getAPIManagerConfigurationService().getAPIManagerConfiguration().getGatewayArtifactSynchronizerProperties().
+                setArtifactSynchronizerDataSource("java:/comp/env/jdbc/WSO2AM_DB");
+        GatewayArtifactsMgtDBUtil.initialize();
+        gatewayArtifactsMgtDAO = GatewayArtifactsMgtDAO.getInstance();
+    }
+
+
+    private static void initializeDatabase(String configFilePath) {
+
+        InputStream in;
+        try {
+            in = FileUtils.openInputStream(new File(configFilePath));
+            StAXOMBuilder builder = new StAXOMBuilder(in);
+            String dataSource = builder.getDocumentElement().getFirstChildWithName(new QName("DataSourceName")).
+                    getText();
+            OMElement databaseElement = builder.getDocumentElement().getFirstChildWithName(new QName("Database"));
+            String databaseURL = databaseElement.getFirstChildWithName(new QName("URL")).getText();
+            String databaseUser = databaseElement.getFirstChildWithName(new QName("Username")).getText();
+            String databasePass = databaseElement.getFirstChildWithName(new QName("Password")).getText();
+            String databaseDriver = databaseElement.getFirstChildWithName(new QName("Driver")).getText();
+
+            BasicDataSource basicDataSource = new BasicDataSource();
+            basicDataSource.setDriverClassName(databaseDriver);
+            basicDataSource.setUrl(databaseURL);
+            basicDataSource.setUsername(databaseUser);
+            basicDataSource.setPassword(databasePass);
+
+            // Create initial context
+            System.setProperty(Context.INITIAL_CONTEXT_FACTORY,
+                    "org.apache.naming.java.javaURLContextFactory");
+            System.setProperty(Context.URL_PKG_PREFIXES,
+                    "org.apache.naming");
+            try {
+                InitialContext.doLookup("java:/comp/env/jdbc/WSO2AM_DB");
+            } catch (NamingException e) {
+                InitialContext ic = new InitialContext();
+                ic.createSubcontext("java:");
+                ic.createSubcontext("java:/comp");
+                ic.createSubcontext("java:/comp/env");
+                ic.createSubcontext("java:/comp/env/jdbc");
+
+                ic.bind("java:/comp/env/jdbc/WSO2AM_DB", basicDataSource);
+            }
+        } catch (XMLStreamException e) {
```

Comment:
All exceptions should be thrown in this scenario

Reply:
https://github.com/wso2/carbon-apimgt/pull/9002/commits/ca3a99296eb9ea796869ee228c7e76f0e38123a7